cumulusci/tasks/bulkdata/extract.py

```diff
@@ -286,6 +290,13 @@ def _create_table(self, mapping):
         mapper_kwargs = {}
         self.models[mapping["table"]] = type(model_name, (object,), {})
 
+        # Track IsPersonAccount for Account and Contact tables if person accounts is enabled.
+        # IsPersonAccount field should never exist in a mapping because the field is not creatable.
+        if (
+            mapping["sf_object"].lower() in ["account", "contact"]
+        ) and self._is_person_accounts_enabled():
+            mapping["fields"]["IsPersonAccount"] = "IsPersonAccount"
```

Comment:
Open to discussion on the best place, but I'd prefer to have this kind of mapping mutation go in the `mapping_parser` module along with the FLS checker and namespace injector. (Those have been added since you started this work). Keeping as much of that logic in one place as possible will help reduce the level of magic required to work on this code and let us keep `_create_table` minimal.

Reply:
This seems like a great approach.  Will dig into change request.

Reply:
This is done.

Reply:
@spelak-salesforce The new validation code in mapping_parser.py does its own describe call for each object in order to validate FLS and inject namespaces. Now that that is happening, that would be a natural place to check if Account and Contact have the IsPersonAccount field and update the mapping accordingly. The actual tasks could then use presence of IsPersonAccount in the mapping fields as a trustworthy indicator for whether the org has person accounts enabled, and we could get rid of OrgInfoMixin and the _org_has_person_accounts_enabled method.

But -- things are working as is, the release is soon, and I don't think it's worth holding this up to do that refactoring. It can happen later.